eval.ml

```diff
@@ -94,11 +95,19 @@ let value_eq (v1 : value) (v2 : value) =
   | Int n1, Int n2 -> n1 = n2
   | Int _, _ | _, Int _ -> false
   | Fun_ref f1, Fun_ref f2 -> f1 = f2
+  | Fun_ref _, _ | _, Fun_ref _ -> false
+  | Array vs1, Array vs2 ->
```

Comment:
I don't know of other patterns that are redundant -- in general redundant patterns are programming errors so it's good to heed the warnings and remove them. This idiom is the only one I know where it is natural to write a redundant pattern. If we want to have the last case, I would rather have it commented out to not get the warning. Otherwise I think that expecting the next person that adds a new case to recognize the pattern and do the right thing is not unreasonable.